code/src/layouts/app-header/AppHeader.test.jsx

```diff
@@ -0,0 +1,18 @@
+import { render } from "@testing-library/react";
+import AppHeader from "@/layouts/app-header/AppHeader";
+import Header from "@/layouts/app-header/components/header/Header";
+import Categories from "@/layouts/app-header/components/categories/Categories";
+
+jest.mock("@/layouts/app-header/components/header/Header", () => () => (
+  <div>Header</div>

```

Comment:
`<div>` is redundant because you search only by text. Maybe it can improve performance

Reply:
> `<div>` is redundant because you search only by text. Maybe it can improve performance

<img width="925" alt="image" src="https://github.com/idx-academy/spa-orders/assets/56305508/e5eaa6d6-7d43-49b8-92ff-21dc709ce8bf">

Reply:
Why we need to use `() => () =>?` I don't get the point

Reply:
> () => () =>?

This approach is useful in testing because it allows you to replace complex components with simple mock implementations, making it easier to test other parts of your application without worrying about the internal details of the mocked component

I have a good article about HOF
FYI -> https://www.freecodecamp.org/news/higher-order-functions-explained/